extensions/spark/kyuubi-spark-lineage/src/test/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParserHelperSuite.scala

```diff
@@ -690,6 +690,122 @@ class SparkSQLLineageParserHelperSuite extends KyuubiFunSuite
     }
   }
 
+  test("columns lineage extract - subquery sql") {
+    val ddls =
+      """
+        |create table table0(a int, b string, c string)
+        |create table table1(a int, b string, c string)
+        |""".stripMargin
+    ddls.split("\n").filter(_.nonEmpty).foreach(spark.sql(_).collect())
+    withTable("table0", "table1") { _ =>
+      val sql0 =
+        """
+          |select a as aa, bb, cc from (select b as bb, c as cc from table1) t0, table0
+          |""".stripMargin
+      val ret0 = exectractLineage(sql0)
+      assert(ret0 == Lineage(
+        List("default.table0", "default.table1"),
+        List(),
+        List(
+          ("aa", Set("default.table0.a")),
+          ("bb", Set("default.table1.b")),
+          ("cc", Set("default.table1.c")))))
+
+      // ScalarSubquery
+      val sql1 =
+        """
+          |select (select a from table1) as aa, b as bb from table1
+          |""".stripMargin
+      val ret1 = exectractLineage(sql1)
+      assert(ret1 == Lineage(
+        List("default.table1"),
+        List(),
+        List(
+          ("aa", Set("default.table1.a")),
+          ("bb", Set("default.table1.b")))))
+
+      val sql2 =
+        """
+          |select (select count(*) from table1) as aa, b as bb from table1
+          |""".stripMargin
+      val ret2 = exectractLineage(sql2)
+      assert(ret2 == Lineage(
+        List("default.table1"),
+        List(),
+        List(
+          ("aa", Set()),
```

Comment:
interesting, the input lineage of `select count(*) from table1` is empty ? how about `table1.__all__` ?

Reply:
1. The definition of column lineage should be column-to-column, which would be ambiguous.
2. IMO, the generation of `count(*)` should have nothing to do with columns. like the following sql
```
select count(*) from (select a from t0) t
select count(*) from (select a, b from t0) t
```
They come from different columns and do have the same results

Reply:
can you change the test case to `select (select count(*) from table0) as aa, b as bb from table1` ?

Reply:
ok，i wiil try it